### PR TITLE
Follow up on #3503 (pass service env vars to PAM sessions)

### PR DESCRIPTION
**Pull request by joukewitteveen**
Errors observed by @evverx inspired these two minor changes, but to be honest I don't think the presented backtraces point at #3503 as their cause.

### CODE REVIEW

**Comment by joukewitteveen** [comment]
While `strv_free` sets the individual freed strings to `NULL`, it (of course) does not do the same to the vector itself. This is my mistake in understanding and I now feel that the second commit of this PR actually fixes the problems identified by @evverx.

The first commit is unrelated to the freeing issues, but is relevant since `pam_putenv` overwrites variables when multiple definitions are encountered, whereas `systemd` wants to stick with the first definition.

File: src/core/execute.c [hunk 1]
```diff
@@ -872,8 +872,10 @@ static int setup_pam(
 
         STRV_FOREACH(e, *env) {
                 pam_code = pam_putenv(handle, *e);
-                if (pam_code != PAM_SUCCESS)
+                if (pam_code != PAM_SUCCESS) {
+                        e = NULL;
```
Comment (evverx) [inline src/core/execute.c]: Indeed, this fixes the issue. Thanks!
But `e=NULL` is tricky. I think we should introduce a new variable (as you suggested: https://github.com/systemd/systemd/pull/3503#discussion_r69868046)

Comment (joukewitteveen) [inline src/core/execute.c]: A similar pattern occurs slightly [above](https://github.com/systemd/systemd/blob/master/src/core/execute.c#L863) this code, but if you rather see a `char **nv` added, that is okay too. I'll update the PR later today.

**Comment by evverx** [comment]
Thanks! Works fine.

``` sh
user@0.service: Child 236 belongs to user@0.service
user@0.service: Main process exited, code=exited, status=224/PAM
user@0.service: Changed start -> dead
user@0.service: Job user@0.service/start finished, result=done
```

instead of

``` sh
user@0.service: Child 237 belongs to user@0.service
user@0.service: Main process exited, code=dumped, status=6/ABRT
user@0.service: Changed start -> dead
user@0.service: Job user@0.service/start finished, result=done
```

I wonder why I don't see [`PAM failed...`](https://github.com/systemd/systemd/blob/f4170c671b863a211056972a469abd416086f22c/src/core/execute.c#L1004). But this is another issue (and seems like forgotten `log_open`, ..., `log_close`)

**Pull request merged**